### PR TITLE
[CELEBORN-638] Migrate configurations celeborn.ha.master.* to celeborn.master.ha.*

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

It was discussed during the last meeting, but abandoned due to the complication.

### Why are the changes needed?

Make the configuration unified.

### Does this PR introduce _any_ user-facing change?

Yes, but the legacy configurations still take effect.

### How was this patch tested?

New UTs.

### CODE REVIEW

**Comment by pan3793** [comment]
@AngersZhuuuu @waitinfuture Before adding UT, I'm wondering if we have reached a consensus prefer to use the `celeborn.master.ha` namespace rather than `celeborn.ha.master`.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -534,32 +534,43 @@ class CelebornConf(loadDefaults: Boolean) extends Cloneable with Logging with Se
...
 
     val nodeConfPrefix = extractPrefix(HA_MASTER_NODE_HOST.key, "<id>")
-    getAllWithPrefix(nodeConfPrefix)
+    val nodeIds = getAllWithPrefix(nodeConfPrefix)
       .map(_._1)
       .filterNot(_.equals("id"))
       .map(k => extractPrefix(k, "."))
       .distinct
+
+    val legacyNodeConfPrefix = extractPrefix(HA_MASTER_NODE_HOST.alternatives.head._1, "<id>")
```
[the remark this answers is not in the document]
Comment (pan3793) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:543]: The assumption is true if #1547 gets in.

**Comment by AngersZhuuuu** [comment]
> @AngersZhuuuu @waitinfuture Before adding UT, I'm wondering if we have reached a consensus prefer to use the `celeborn.master.ha` namespace rather than `celeborn.ha.master`.

For me, `celeborn.master.ha` is more better

**Comment by pan3793** [comment]
@FMX @waitinfuture any thoughts on this change?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (waitinfuture) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:543]: Better to add a comment to explain a bit.

**Review by waitinfuture — APPROVED** [review]
LGTM except a nit, thanks!

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (pan3793) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:544]: done

**Comment by waitinfuture** [comment]
Thanks, merging to main/0.3

**Pull request closed without merging**